src/Core/Form/IdentifiableObject/CommandBuilder/Accessor/CommandAccessorConfig.php

```diff
@@ -0,0 +1,120 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Form\IdentifiableObject\CommandBuilder\Accessor;
+
+/**
+ * Config that gives information to the CommandAccessor component to correctly check the data and prefill the command
+ * object, each data property is associated to a setter in the command and its appropriate type for casting. Example:
+ *
+ * $config = new CommandAccessorConfig('modify_all_');
+ * $config
+ *     ->addField('[name]', 'setName', CommandField::TYPE_STRING)
```

Comment:
Isn't more obvious to add a field like that 
```suggestion
 *     ->addField('name', 'setName', CommandField::TYPE_STRING)
```
or
```suggestion
 *     ->addField('parent[children]', 'setChildren', CommandField::TYPE_STRING)
```

Reply:
No, I thought so as well, but that's not how the `PropertyAccessor` works. When you want to access an array field you need those brackets. The notation without brackets seems to be reserved for Objects like in `object.myChildren`

I tested the notation `parent[children]` and I get an exception `Symfony\Component\PropertyAccess\Exception\NoSuchPropertyException: Cannot read property "parent" from an array. Maybe you intended to write the property path as "[parent]" instead.`

Reply:
Actually I think it would work if you had this kind of object:
```php
class MyObject {
    public $parent = [
        'children' => 'anything',
    ];
}
```

It mixed both object and array so that would match your example. But since this component will mainly be used in command builders which are based on flat array data from the form (no objects), I prefer to keep the examples matching the most common use cases.

Reply:
sad story :(

Reply:
I thought so too at first, but in a way it makes sense that they chose to use a different syntax to access arrays and objects

Reply:
I encountered the same situation in https://github.com/PrestaShop/PrestaShop/pull/25611. I agree it feels strange at first.